dm/master/shardddl/optimist.go

```diff
@@ -104,6 +109,51 @@ func (o *Optimist) Locks() map[string]*optimism.Lock {
 	return o.lk.Locks()
 }
 
+// ShowLocks is used by `show-ddl-locks` command.
+func (o *Optimist) ShowLocks(task string, sources []string) []*pb.DDLLock {
```

Comment:
do we need to add read lock?

Reply:
use read lock to protect what?

Reply:
ok, a mistake, the `o.lk.locks` returns a copy, don't need to protect